oem/ibm/libpldmresponder/file_io.cpp

```diff
@@ -657,10 +710,12 @@ Response rwFileByTypeIntoMemory(uint8_t cmd, const pldm_msg* request,
         return response;
     }
 
-    std::unique_ptr<FileHandler> handler{};
+    std::shared_ptr<FileHandler> handler{};
+    sdeventplus::Event event = sdeventplus::Event::get_default();
     try
     {
-        handler = getHandlerByType(fileType, fileHandle);
+        handler = getSharedHandlerByType(fileType, fileHandle);
```

Comment:
Why are we using shared pointer? Once the file handler is created, we are supposed to save it and resuse it. 
Can we use same gethandlerByType function to return unique pointer and then move the pointer  to functionPtr?

Reply:
when we call any function of read/write for DMA transfer then first time it's create event loop and return from that function and delete unique_ptr because of limited scope. we can't move unique_ptr inside event loop because it's delete any point of time and we don't know, and while creating even loop need to pass function pointer which will be called after file transfer completed and do post operation. so as per our requirement shared_ptr is best to use.